pages/dashboard/DisplayInvitations.js

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useRouter } from 'next/router';
+
+import { ServiceTable } from '../../components/UI/ServiceTable';
+import TextButton from '../../components/UI/TextButton';
+import AcceptIcon from '../../assets/icons/accept.svg';
+import CloseIcon from '../../assets/icons/close.svg';
+import LoadingSpinnerInline from '../../components/UI/LoadingSpinnerInline';
+
+/**
+ * COMPONENT 'DisplayInvitations'
+ * Displays an invitation for a matrix room/space within a <ServiceTable> component.
+ *
+ * @param {Object} invite — object of the room the user was invited to
+ * @param {String} service — name of the service (i.e. the object key names of the 'authProvider' object in the config)
+ * @param {String} name — name of the Application (i.e. the 'path' variable in the config)
+ *
+ * @callback acceptMatrixInvite - callback function to accept invitations to a matrix room or space
+ * @param e - event triggered by the button
+ * @param roomId - matrix roomId
+ * @param service - name of the service (parsed to the function)
+ *
+ * @callback declineMatrixInvite - callback function to decline invitations to a matrix room or space
+ * @param e - event triggered by the button
+ * @param roomId - matrix roomId
+ *
+ * @returns {React.ReactComponent}
+*/
+
+const DisplayInvitations = ({ invite, name, service, acceptMatrixInvite, declineMatrixInvite }) => {
+    const { t } = useTranslation('dashboard');
+    const [isAcceptingInvite, setIsAcceptingInvite] = useState(false);
+    const [isDecliningInvite, setisDecliningInvite] = useState(false);
+    const router = useRouter();
+
+    const handleDecline = async (e, roomId) => {
+        setisDecliningInvite(true);
+        await declineMatrixInvite(e, roomId);
```

Comment:
Instead of forwarding the `e` only to prevent the default click handler logic, please do that here.
As a rule of thumb: Catch those default behaviors always as close as possible to the original sending element. In this case it's the table cell. And not something that's defined in the file where you're currently calling `e.preventDefault()`. This will also allow you to not having to pass the `e` down the call stack all the way.

Reply:
(Same goes for `acceptMatrixInvite` below)